_rules/heading-accessible-name-ffd0e9.md

```diff
@@ -0,0 +1,153 @@
+---
+id: ffd0e9
+name: Heading has accessible name
+rule_type: atomic
+description: |
+  This rule checks that each heading has an accessible name.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag20:2.4.6: # Headings and Labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any HTML element with the [semantic role][] of `heading` that is [included in the accessibility tree][].
+
+## Expectation
+
+Each test target has a non-empty (`""`) [accessible name][].
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+Some assistive technologies may hide headings with empty [accessible name][] from the users. This depends both on the user agent and how the [accessible name][] was computed (the [accessible name and description computation][] is not clear concerning which characters should be trimmed) and of the assistive technology itself. Hence, there are cases where the outcome of this rule is _failed_, but users of certain assistive technology and browser combinations will not experience an issue.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Info and Relationships][usc131]
+- [Understanding Success Criterion 2.4.6: Headings and Labels][usc246]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `h1` element has a non-empty [accessible name][].
+
+```html
+<h1>ACT rules</h1>
+```
+
+#### Passed Example 2
+
+This `div` element has a [semantic role][] of heading and a non-empty [accessible name][].
+
+```html
+<div role="heading" aria-level="1">ACT rules</div>
```

Comment:
The aria-level attribute is not required in an example for this rule.

Reply:
I've been followimg @WilcoFiers's advice on this:
https://github.com/act-rules/act-rules.github.io/pull/400#discussion_r321177593

I think the idea is that passing examples shouldn't be failing the SC and that a `div role="heading">Foo</div>` does pass the rule but fails the SC for lack of full structure information (i.e. heading level).